core/vote/vote_pool.go

```diff
@@ -0,0 +1,397 @@
+package vote
+
+import (
+	"container/heap"
+	"sync"
+
+	mapset "github.com/deckarep/golang-set"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/consensus"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+const (
+	maxCurVoteAmountPerBlock    = 21
```

Comment:
This must be greater or equal to 22, not sure if we want more than 22 validators in the future.

Reply:
we should move this to config